test/e2e/service_test.go

```diff
@@ -106,38 +106,50 @@ func (data *TestData) createClusterIPServiceAndBackendPods(t *testing.T, name st
 	return svc, cleanup
 }
 
-// TestNodePortWindows tests NodePort Service on Windows Node. It is a temporary test to replace upstream Kubernetes one:
-// https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L42
-// Issue: https://github.com/antrea-io/antrea/issues/2289
-func TestNodePortWindows(t *testing.T) {
-	skipIfNoWindowsNodes(t)
-
+// TestNodePort tests traffic from Pods to NodePort Service.
+func TestNodePort(t *testing.T) {
 	data, err := setupTest(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
 
+	nodeIP := clusterInfo.nodes[0].ip()
+	if net.ParseIP(nodeIP).To4() == nil {
+		nodeIP = fmt.Sprintf("[%s]", nodeIP)
+	}
+	t.Run("Windows NodePort Service", func(t *testing.T) {
+		// Windows NodePort Service: It is a temporary test to replace upstream Kubernetes one:
+		// https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L42
+		// Issue: https://github.com/antrea-io/antrea/issues/2289
+		// Unlike upstream Kubernetes Conformance, the client is on a Linux Node (nodeName(0)).
+		// It doesn't need to be the control-plane for e2e test and other Linux workers will work as well. However, in this
+		// e2e framework, nodeName(0)/Control-plane Node is guaranteed to be a Linux one.
+		skipIfNoWindowsNodes(t)
+		svcNode := nodeName(clusterInfo.windowsNodes[0])
+		testNodePort(t, data, svcNode, "windows-service-test", nodeIP)
+	})
+	t.Run("Linux NodePort Service", func(t *testing.T) {
+		svcNode := clusterInfo.nodes[0].name
+		testNodePort(t, data, svcNode, "linux-service-test", nodeIP)
+	})
```

Comment:
I'm curious: is this not covered by upstream tests?

Reply:
We are skipping both linux and windows nodeport service tests. Let me use this PR to verify if they work now.
https://github.com/antrea-io/antrea/blob/4f7c6ebb6c8d63af1ceb1b6bc6513ee17349ba77/ci/jenkins/test.sh#L35